### ISSUE TITLE
Intermittent test failures on CI

### ISSUE THREAD

**Issue by chrisroos** [alphagov/manuals-publisher#898]
We feel as though we've seen intermittent failures on CI so I'm opening this issue to allow us to record information about when they happen.

**Comment by chrisroos** [comment]
[Build 792 of master failed on CI](https://ci.integration.publishing.service.gov.uk/job/manuals-publisher/job/master/792/console). All the tests pass when run with the same seed locally:

```
/Users/chrisroos/.rbenv/versions/2.2.4/bin/ruby \
-I/Users/chrisroos/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/lib:/Users/chrisroos/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/rspec-support-3.4.1/lib \
/Users/chrisroos/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/rspec-core-3.4.4/exe/rspec \
--pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb \
--seed 50719
```

I generated the command above by running `bundle exec rake` on my laptop and adding the `--seed` option.

**Comment by floehopper** [comment]
I think Jenkins is only keeping 10 master builds around, so I've [captured the output for build 792 in a gist](https://gist.github.com/floehopper/63dbb75272074436feae00f5d6331ee4) for posterity.

**Comment by floehopper** [comment]
From @gpeng:

> may be totally unrelated but if you get two builds going on the same agent at the moment and haven’t namespaced your DBs then on drops the db from the other. Not sure that’s relevant here with mongo. Happens with postgres and mysql though. https://github.com/alphagov/whitehall/blob/master/config/database.yml#L14

**Comment by floehopper** [comment]
I think there might also be another problem. I can't see any evidence (in code or CI build output) that **any** database reset is happening at all in the CI build. Neither the [`setupDb` function in the app's `Jenkinsfile`](https://github.com/alphagov/manuals-publisher/blob/45e9fda273c4165c5f667c47a0dfbf12727db1a4/Jenkinsfile#L6-L11) nor the [`setupDb` function in `govuk-puppet`'s `govuk_jenkinslib.groovy` ](https://github.com/alphagov/govuk-puppet/blob/ddf8215844825dcad5402ea9986b119016957d5c/modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy#L191-L199) seem to be being called. However, given [this comment](https://github.com/alphagov/manuals-publisher/blob/45e9fda273c4165c5f667c47a0dfbf12727db1a4/Jenkinsfile#L6-L7), perhaps I'm missing something...?

**Comment by gpeng** [comment]
My comment up there ^ may be unrelated. This was from a discussion in slack earlier: 

> If there are any build steps which shouldn't be run on the same node in parallel (e.g. tests that access the same DB) you can add a `lock` around that step: https://github.com/alphagov/whitehall/pull/3121/files

again... may be totally unrelated.

**Comment by chrisroos** [comment]
I saw another test failure today - https://ci.integration.publishing.service.gov.uk/job/manuals-publisher/job/improve-change-note-validator/2/consoleText.

I've pasted the [output to this gist for reference](https://gist.github.com/chrisroos/ff19382360e8b8893e8b0cf79dbf83d4).

**Comment by chrisroos** [comment]
I noticed that the deployed-to-production branch build 495 failed on Jenkins _after_ PR #959 was merged. Having investigated I can see that the deployed-to-production branch didn't include the fix from PR #959 so I'm happy to leave this issue closed.

**Comment by chrisroos** [comment]
We're still seeing quite a few test failures on CI so I'm reopening this issue.

We've started investigating the issues in PR #974.
